pkg/scale/result.go

```diff
@@ -0,0 +1,41 @@
+package scale
+
+import (
+	"fmt"
+	"reflect"
+)
+
+type resultCache map[string]map[bool]interface{}
+type Result struct {
+	Ok  interface{}
+	Err interface{}
+}
+
+// Validate ensures Result is valid.  Only one of the Ok and Err attributes should be nil
+func (r Result) Validate() (err error) {
+	switch {
+	case r.Ok == nil && r.Err != nil, r.Ok != nil && r.Err == nil:
+	default:
+		err = fmt.Errorf("Result is invalid: %+v", r)
+	}
+	return
+}
+
+var resCache resultCache = make(resultCache)
+
+func RegisterResult(in interface{}, inOK interface{}, inErr interface{}) (err error) {
```

Comment:
add comment pls

Reply:
also, can `inOk` or `inErr` be `nil`? so that encoding of `Ok` is just `[]byte{0}` and encoding of `Err` is just `[]byte{1}`? can you add a test case for that?

Reply:
TODO @timwu20

Reply:
let me know when this is done and I can re-review, this should be the last thing to update :D